diff -ur gts-0.7.6.orig/examples/Makefile.am gts-0.7.6/examples/Makefile.am
--- gts-0.7.6.orig/examples/Makefile.am	2005-03-09 03:18:56.000000000 +0100
+++ gts-0.7.6/examples/Makefile.am	2006-05-19 14:30:04.000000000 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src -I$(includedir)\
+INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src\
 	 -DG_LOG_DOMAIN=\"Gts-examples\"
 LDADD = $(top_builddir)/src/libgts.la -lm
 DEPS = $(top_builddir)/src/libgts.la
diff -ur gts-0.7.6.orig/examples/Makefile.in gts-0.7.6/examples/Makefile.in
--- gts-0.7.6.orig/examples/Makefile.in	2006-03-10 01:31:05.000000000 +0100
+++ gts-0.7.6/examples/Makefile.in	2006-05-19 14:30:38.000000000 +0200
@@ -293,7 +293,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src -I$(includedir)\
+INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src\
 	 -DG_LOG_DOMAIN=\"Gts-examples\"
 
 LDADD = $(top_builddir)/src/libgts.la -lm
diff -ur gts-0.7.6.orig/src/Makefile.am gts-0.7.6/src/Makefile.am
--- gts-0.7.6.orig/src/Makefile.am	2005-01-19 05:48:22.000000000 +0100
+++ gts-0.7.6/src/Makefile.am	2006-05-19 14:30:04.000000000 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-INCLUDES = -I$(top_srcdir) -I$(includedir) -DG_LOG_DOMAIN=\"Gts\"
+INCLUDES = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Gts\"
 
 bin_SCRIPTS=gts-config
 
diff -ur gts-0.7.6.orig/src/Makefile.in gts-0.7.6/src/Makefile.in
--- gts-0.7.6.orig/src/Makefile.in	2006-03-10 01:31:05.000000000 +0100
+++ gts-0.7.6/src/Makefile.in	2006-05-19 14:30:38.000000000 +0200
@@ -222,7 +222,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-INCLUDES = -I$(top_srcdir) -I$(includedir) -DG_LOG_DOMAIN=\"Gts\"
+INCLUDES = -I$(top_srcdir) -DG_LOG_DOMAIN=\"Gts\"
 bin_SCRIPTS = gts-config
 BUILT_SOURCES = \
 	gts-config \
diff -ur gts-0.7.6.orig/test/boolean/Makefile.am gts-0.7.6/test/boolean/Makefile.am
--- gts-0.7.6.orig/test/boolean/Makefile.am	2005-01-19 05:47:06.000000000 +0100
+++ gts-0.7.6/test/boolean/Makefile.am	2006-05-19 14:30:04.000000000 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src -I$(includedir)\
+INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src\
 	 -DG_LOG_DOMAIN=\"Gts-test\"
 LDADD = $(top_builddir)/src/libgts.la -lm
 DEPS = $(top_builddir)/src/libgts.la
diff -ur gts-0.7.6.orig/test/boolean/Makefile.in gts-0.7.6/test/boolean/Makefile.in
--- gts-0.7.6.orig/test/boolean/Makefile.in	2006-03-10 01:31:06.000000000 +0100
+++ gts-0.7.6/test/boolean/Makefile.in	2006-05-19 14:30:38.000000000 +0200
@@ -198,7 +198,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src -I$(includedir)\
+INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src\
 	 -DG_LOG_DOMAIN=\"Gts-test\"
 
 LDADD = $(top_builddir)/src/libgts.la -lm
diff -ur gts-0.7.6.orig/test/coarsen/Makefile.am gts-0.7.6/test/coarsen/Makefile.am
--- gts-0.7.6.orig/test/coarsen/Makefile.am	2005-01-19 05:47:06.000000000 +0100
+++ gts-0.7.6/test/coarsen/Makefile.am	2006-05-19 14:30:04.000000000 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src -I$(includedir)\
+INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src\
 	 -DG_LOG_DOMAIN=\"Gts-test\"
 LDADD = $(top_builddir)/src/libgts.la -lm
 DEPS = $(top_builddir)/src/libgts.la
diff -ur gts-0.7.6.orig/test/coarsen/Makefile.in gts-0.7.6/test/coarsen/Makefile.in
--- gts-0.7.6.orig/test/coarsen/Makefile.in	2006-03-10 01:31:06.000000000 +0100
+++ gts-0.7.6/test/coarsen/Makefile.in	2006-05-19 14:30:38.000000000 +0200
@@ -208,7 +208,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src -I$(includedir)\
+INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src\
 	 -DG_LOG_DOMAIN=\"Gts-test\"
 
 LDADD = $(top_builddir)/src/libgts.la -lm
diff -ur gts-0.7.6.orig/test/delaunay/Makefile.am gts-0.7.6/test/delaunay/Makefile.am
--- gts-0.7.6.orig/test/delaunay/Makefile.am	2005-01-19 05:47:06.000000000 +0100
+++ gts-0.7.6/test/delaunay/Makefile.am	2006-05-19 14:30:04.000000000 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src -I$(includedir) \
+INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src \
 	 -DG_LOG_DOMAIN=\"Gts-test\"
 LDADD = $(top_builddir)/src/libgts.la -lm
 DEPS = $(top_builddir)/src/libgts.la
diff -ur gts-0.7.6.orig/test/delaunay/Makefile.in gts-0.7.6/test/delaunay/Makefile.in
--- gts-0.7.6.orig/test/delaunay/Makefile.in	2006-03-10 01:31:06.000000000 +0100
+++ gts-0.7.6/test/delaunay/Makefile.in	2006-05-19 14:30:38.000000000 +0200
@@ -193,7 +193,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src -I$(includedir) \
+INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src \
 	 -DG_LOG_DOMAIN=\"Gts-test\"
 
 LDADD = $(top_builddir)/src/libgts.la -lm
diff -ur gts-0.7.6.orig/tools/Makefile.am gts-0.7.6/tools/Makefile.am
--- gts-0.7.6.orig/tools/Makefile.am	2005-01-19 05:47:06.000000000 +0100
+++ gts-0.7.6/tools/Makefile.am	2006-05-19 14:30:04.000000000 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src -I$(includedir)\
+INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src\
 	 -DG_LOG_DOMAIN=\"Gts-tools\"
 LDADD = $(top_builddir)/src/libgts.la -lm
 DEPS = $(top_builddir)/src/libgts.la
diff -ur gts-0.7.6.orig/tools/Makefile.in gts-0.7.6/tools/Makefile.in
--- gts-0.7.6.orig/tools/Makefile.in	2006-03-10 01:31:07.000000000 +0100
+++ gts-0.7.6/tools/Makefile.in	2006-05-19 14:30:38.000000000 +0200
@@ -221,7 +221,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src -I$(includedir)\
+INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/src\
 	 -DG_LOG_DOMAIN=\"Gts-tools\"
 
 LDADD = $(top_builddir)/src/libgts.la -lm
