diff -uNr gts-0.7.6.orig/gts.pc.in gts-0.7.6/gts.pc.in
--- gts-0.7.6.orig/gts.pc.in	2005-01-19 05:48:19.000000000 +0100
+++ gts-0.7.6/gts.pc.in	2007-10-21 08:57:59.000000000 +0200
@@ -6,6 +6,6 @@
 Name: GTS
 Description: GNU Triangulated Surface Library
 Version: @VERSION@
-Requires: glib-2.0,gthread-2.0,gmodule-2.0
-Libs: -L${libdir} @LIBS@ -lgts
+Requires: glib-2.0
+Libs: -L${libdir} @LIBS@ -lgts -lm
 Cflags: -I${includedir}
diff -uNr gts-0.7.6.orig/src/gts-config.in gts-0.7.6/src/gts-config.in
--- gts-0.7.6.orig/src/gts-config.in	2005-01-19 05:47:06.000000000 +0100
+++ gts-0.7.6/src/gts-config.in	2007-10-21 09:22:02.000000000 +0200
@@ -1,14 +1,5 @@
 #!/bin/sh
 
-glib_libs="@glib_libs@"
-glib_cflags="@glib_cflags@"
-glib_thread_libs="@glib_thread_libs@"
-glib_thread_cflags="@glib_thread_cflags@"
-glib_module_libs="@glib_module_libs@"
-glib_module_cflags="@glib_module_cflags@"
-
-prefix=@prefix@
-exec_prefix=@exec_prefix@
 exec_prefix_set=no
 
 usage()
@@ -30,24 +21,13 @@
 	exit $1
 }
 
-uniquify()
-{
-    echo $1 | awk '{
-	for (i = 1; i <= NF; i++) {
-	    unique = 1;
-	    for (j = i + 1; j <= NF && unique; j++)
-		if ($i == $j)
-		    unique = 0;
-            if (unique)
-		printf ("%s ", $i);
-	}
-    }'
-}
-
 if test $# -eq 0; then
 	usage 1 1>&2
 fi
 
+PKG_CONFIG_ARGS=
+PKG_CONFIG_MODS=gts
+
 while test $# -gt 0; do
   case "$1" in
   -*=*) optarg=`echo "$1" | sed 's/[-_a-zA-Z0-9]*=//'` ;;
@@ -56,23 +36,23 @@
 
   case $1 in
     --prefix=*)
-      prefix=$optarg
+      PKG_CONFIG_ARGS="$PKG_CONFIG_ARGS --define-variable=prefix=$optarg"
       if test $exec_prefix_set = no ; then
-        exec_prefix=$optarg
+        PKG_CONFIG_ARGS="$PKG_CONFIG_ARGS --define-variable=exec_prefix=$optarg"
       fi
       ;;
     --prefix)
       echo_prefix=yes
       ;;
     --exec-prefix=*)
-      exec_prefix=$optarg
+      PKG_CONFIG_ARGS="$PKG_CONFIG_ARGS --define-variable=exec_prefix=$optarg"
       exec_prefix_set=yes
       ;;
     --exec-prefix)
       echo_exec_prefix=yes
       ;;
     --version)
-      echo @GTS_VERSION@
+      pkg-config --modversion gts
       exit 0
       ;;
     --cflags)
@@ -85,13 +65,12 @@
       echo_check=yes
       ;;
     gts)
-      lib_gts=yes
       ;;
     gmodule)
-      lib_gmodule=yes
+      PKG_CONFIG_MODS="$PKG_CONFIG_MODS gmodule-2.0"
       ;;
     gthread)
-      lib_gthread=yes
+      PKG_CONFIG_MODS="$PKG_CONFIG_MODS gthread-2.0"
       ;;
     *)
       usage 1 1>&2
@@ -101,34 +80,18 @@
 done
 
 if test "$echo_prefix" = "yes"; then
-	echo $prefix
+	pkg-config $PKG_CONFIG_ARGS --variable=prefix gts
 fi
 if test "$echo_exec_prefix" = "yes"; then
-	echo $exec_prefix
+	pkg-config $PKG_CONFIG_ARGS --variable=exec_prefix gts
 fi
 if test "$echo_cflags" = "yes"; then
-	if test "$lib_gmodule" = "yes"; then
-	    glib_cflags="$glib_cflags $glib_module_cflags"
-	fi
-	if test "$lib_gthread" = "yes"; then
-	    glib_cflags="$glib_cflags $glib_thread_cflags"
-	fi
-	glib_cflags="$glib_cflags -I${prefix}/include"
-	glib_cflags=`uniquify "$glib_cflags"`
         if test "$echo_check" = "yes"; then
-	    echo -DGTS_CHECK_CASTS $glib_cflags
+           echo -DGTS_CHECK_CASTS `pkg-config $PKG_CONFIG_ARGS --cflags $PKG_CONFIG_MODS`
         else
-	    echo $glib_cflags
-	fi
+           echo `pkg-config $PKG_CONFIG_ARGS --cflags $PKG_CONFIG_MODS`
+        fi
 fi
 if test "$echo_libs" = "yes"; then
-	if test "$lib_gmodule" = "yes"; then
-	    glib_libs="$glib_libs $glib_module_libs"
-	fi
-	if test "$lib_gthread" = "yes"; then
-	    glib_libs="$glib_libs $glib_thread_libs"
-	fi
-	glib_libs="-L${exec_prefix}/lib -lgts $glib_libs -lm"
-	glib_libs=`uniquify "$glib_libs"`
-	echo $glib_libs
+	echo `pkg-config $PKG_CONFIG_ARGS --libs $PKG_CONFIG_MODS`
 fi
